prometheusremotewrite exporter crashes collector with metrics names < 6 characters

**Describe the bug**
When Collector uses prometheusremotewriteexporter, metric names longer than 6 characters cause it to crash

2021-03-03 09:30:54
panic: runtime error: slice bounds out of range [-1:]

2021-03-03 09:30:54
goroutine 97 [running]:

2021-03-03 09:30:54
go.opentelemetry.io/collector/exporter/prometheusremotewriteexporter.getPromMetricName(0xc0003df7c0, 0x0, 0x0, 0x6, 0x6)

Seems to be caused by this logic which checks for "total" at the end of the metric name: https://github.com/open-telemetry/opentelemetry-collector/blob/68c5961f7bc2d8d55fd431ab07367fc16f8b26d0/exporter/prometheusremotewriteexporter/helper.go#L197

**Steps to reproduce**
Enable Prometheus Remote Write exporter, send OTLP metrics to collector with names shorter than 6 characters

**What version did you use?**
Version: 0.16.0

**What config did you use?**
Config:
extensions:
  health_check:
    port: 13133

receivers:
  otlp:
    protocols:
      grpc:
        endpoint: 0.0.0.0:4317
      http:

processors:
  batch:

exporters:
  logging:
    logLevel: warn
  prometheusremotewrite:
    endpoint: "http://127.0.0.1:9201/write"

service:
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging]

    metrics:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging, prometheusremotewrite]

  extensions: [health_check]

**Environment**
OS: alpine:3.12 (Docker)

--- a/exporter/prometheusremotewriteexporter/helper.go
+++ b/exporter/prometheusremotewriteexporter/helper.go
@@ -31,16 +31,16 @@ import (
 )
 
 const (
-	nameStr     = "__name__"
-	sumStr      = "_sum"
-	countStr    = "_count"
-	bucketStr   = "_bucket"
-	leStr       = "le"
-	quantileStr = "quantile"
-	pInfStr     = "+Inf"
-	totalStr    = "total"
-	delimeter   = "_"
-	keyStr      = "key"
+	nameStr       = "__name__"
+	sumStr        = "_sum"
+	countStr      = "_count"
+	bucketStr     = "_bucket"
+	leStr         = "le"
+	quantileStr   = "quantile"
+	pInfStr       = "+Inf"
+	counterSuffix = "_total"
+	delimiter     = "_"
+	keyStr        = "key"
 )
 
 // ByLabelName enables the usage of sort.Sort() with a slice of labels
@@ -188,13 +188,10 @@ func getPromMetricName(metric *otlp.Metric, ns string) string {
 	b.WriteString(ns)
 
 	if b.Len() > 0 {
-		b.WriteString(delimeter)
+		b.WriteString(delimiter)
 	}
 	name := metric.GetName()
 	b.WriteString(name)
-
-	// do not add the total suffix if the metric name already ends in "total"
-	isCounter = isCounter && name[len(name)-len(totalStr):] != totalStr
 
 	// Including units makes two metrics with the same name and label set belong to two different TimeSeries if the
 	// units are different.
@@ -205,9 +202,12 @@ func getPromMetricName(metric *otlp.Metric, ns string) string {
 		}
 	*/
 
-	if b.Len() > 0 && isCounter {
-		b.WriteString(delimeter)
-		b.WriteString(totalStr)
+	counterSuffixLength := len(counterSuffix)
+	nameLength := len(name)
+	hasCounterSuffix := nameLength > counterSuffixLength && name[nameLength-counterSuffixLength:] == counterSuffix
+
+	if b.Len() > 0 && isCounter && !hasCounterSuffix {
+		b.WriteString(counterSuffix)
 	}
 	return sanitize(b.String())
 }
@@ -262,7 +262,7 @@ func sanitize(s string) string {
 	// See https://github.com/orijtech/prometheus-go-metrics-exporter/issues/4.
 	s = strings.Map(sanitizeRune, s)
 	if unicode.IsDigit(rune(s[0])) {
-		s = keyStr + delimeter + s
+		s = keyStr + delimiter + s
 	}
 	if s[0] == '_' {
 		s = keyStr + s

--- a/exporter/prometheusremotewriteexporter/helper.go
+++ b/exporter/prometheusremotewriteexporter/helper.go
@@ -202,11 +202,7 @@ func getPromMetricName(metric *otlp.Metric, ns string) string {
 		}
 	*/
 
-	counterSuffixLength := len(counterSuffix)
-	nameLength := len(name)
-	hasCounterSuffix := nameLength > counterSuffixLength && name[nameLength-counterSuffixLength:] == counterSuffix
-
-	if b.Len() > 0 && isCounter && !hasCounterSuffix {
+	if b.Len() > 0 && isCounter && !strings.HasSuffix(name, counterSuffix) {
 		b.WriteString(counterSuffix)
 	}
 	return sanitize(b.String())

--- a/exporter/prometheusremotewriteexporter/helper_test.go
+++ b/exporter/prometheusremotewriteexporter/helper_test.go
@@ -287,7 +287,7 @@ func Test_getPromMetricName(t *testing.T) {
 			"total_suffix",
 			validMetrics1[validIntSum],
 			ns1,
-			"test_ns_" + validIntSum + delimeter + totalStr,
+			"test_ns_" + validIntSum + counterSuffix,
 		},
 		{
 			"dirty_string",

--- a/exporter/prometheusremotewriteexporter/helper_test.go
+++ b/exporter/prometheusremotewriteexporter/helper_test.go
@@ -290,6 +290,12 @@ func Test_getPromMetricName(t *testing.T) {
 			"test_ns_" + validIntSum + counterSuffix,
 		},
 		{
+			"already_has_total_suffix",
+			validMetrics1[suffixedCounter],
+			ns1,
+			"test_ns_" + suffixedCounter,
+		},
+		{
 			"dirty_string",
 			validMetrics2[validIntGaugeDirty],
 			"7" + ns1,

--- a/exporter/prometheusremotewriteexporter/testutil_test.go
+++ b/exporter/prometheusremotewriteexporter/testutil_test.go
@@ -94,6 +94,7 @@ var (
 	validIntHistogram    = "valid_IntHistogram"
 	validDoubleHistogram = "valid_DoubleHistogram"
 	validDoubleSummary   = "valid_DoubleSummary"
+	suffixedCounter      = "valid_IntSum_total"
 
 	validIntGaugeDirty = "*valid_IntGauge$"
 
@@ -126,6 +127,18 @@ var (
 		},
 		validIntSum: {
 			Name: validIntSum,
+			Data: &otlp.Metric_IntSum{
+				IntSum: &otlp.IntSum{
+					DataPoints: []*otlp.IntDataPoint{
+						getIntDataPoint(lbs1, intVal1, time1),
+						nil,
+					},
+					AggregationTemporality: otlp.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE,
+				},
+			},
+		},
+		suffixedCounter: {
+			Name: suffixedCounter,
 			Data: &otlp.Metric_IntSum{
 				IntSum: &otlp.IntSum{
 					DataPoints: []*otlp.IntDataPoint{

